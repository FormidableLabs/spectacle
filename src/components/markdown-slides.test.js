import MarkdownSlides from "./markdown-slides";
import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("MarkdownSlides", () => {
  test("should render correctly when using tagged template literals", () => {
    const wrapper = shallow((
      <div>
        {MarkdownSlides`
## Slide 1 Title
---
## Slide 2 Title
        `}
      </div>
    ));
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  test("should render correctly when using tagged template literals and string interpolation", () => {
    const content = "**bold**";
    const wrapper = shallow((
      <div>
        {MarkdownSlides`
## Slide 1 Title
This text is ${content}.
---
## Slide 2 Title
        `}
      </div>
    ));
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  test("should render correctly when using function syntax", () => {
    const markdownContent = `
## Slide A Title
---
## Slide B Title
    `;
    const markdownSlides = MarkdownSlides;
    const wrapper = shallow((
      <div>
        {markdownSlides(markdownContent)}
      </div>
    ));
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
