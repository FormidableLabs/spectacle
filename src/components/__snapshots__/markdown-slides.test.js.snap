exports[`MarkdownSlides should render correctly when using function syntax 1`] = `
<div>
  <Slide
    align="center center"
    presenterStyle={Object {}}
    style={Object {}}
    viewerScaleMode={false}>
    <Markdown
      style={Object {}}>
      
      ## Slide A Title
      
    </Markdown>
  </Slide>
  <Slide
    align="center center"
    presenterStyle={Object {}}
    style={Object {}}
    viewerScaleMode={false}>
    <Markdown
      style={Object {}}>
      
      ## Slide B Title
          
    </Markdown>
  </Slide>
</div>
`;

exports[`MarkdownSlides should render correctly when using tagged template literals 1`] = `
<div>
  <Slide
    align="center center"
    presenterStyle={Object {}}
    style={Object {}}
    viewerScaleMode={false}>
    <Markdown
      style={Object {}}>
      ## Slide 1 Title
      
    </Markdown>
  </Slide>
  <Slide
    align="center center"
    presenterStyle={Object {}}
    style={Object {}}
    viewerScaleMode={false}>
    <Markdown
      style={Object {}}>
      
      ## Slide 2 Title
    </Markdown>
  </Slide>
</div>
`;

exports[`MarkdownSlides should render correctly when using tagged template literals and string interpolation 1`] = `
<div>
  <Slide
    align="center center"
    presenterStyle={Object {}}
    style={Object {}}
    viewerScaleMode={false}>
    <Markdown
      style={Object {}}>
      ## Slide 1 Title
      This text is **bold**.
      
    </Markdown>
  </Slide>
  <Slide
    align="center center"
    presenterStyle={Object {}}
    style={Object {}}
    viewerScaleMode={false}>
    <Markdown
      style={Object {}}>
      
      ## Slide 2 Title
    </Markdown>
  </Slide>
</div>
`;
